File: src/co/there4/mrbean/GenerateCompoundAction.java
package co.there4.mrbean;

import static java.util.Arrays.asList;

import java.util.List;

import com.intellij.psi.PsiClass;
import com.intellij.psi.PsiField;

abstract class GenerateCompoundAction extends GenerateAction {
    private final List<GenerateAction> actions;

    GenerateCompoundAction (String text, String dialogTitle, GenerateAction... actions) {
        super (text, dialogTitle, null);
        this.actions = asList (actions);
    }

    @Override protected void generate (
        PsiClass clazz, List<PsiField> fields) {
        actions.stream ().forEach (action -> action.generate (clazz, fields));
    }
}

File: src/co/there4/mrbean/GenerateCompoundAction.kt
package co.there4.mrbean

import java.util.Arrays.asList

import com.intellij.psi.PsiClass
import com.intellij.psi.PsiField

abstract class GenerateCompoundAction
    internal constructor(text: String, dialogTitle: String, vararg actions: GenerateAction) :
        GenerateAction(text, dialogTitle, null) {

    private val actions: List<GenerateAction> = asList(*actions)

    protected override fun generate(clazz: PsiClass, fields: List<PsiField>) {
        actions.forEach { action -> action.generate(clazz, fields) }
    }
}
